### PR TITLE
[REEF-633] Add a static function to share the getRackName() function …

### PR DESCRIPTION
**Pull request by afchung**
…in YARN

This addressed the issue by
  * Adding ContainerUtilities.getRackName() and replacing original logic with the function call in YarnContainerManager.

JIRA:
  [REEF-633](https://issues.apache.org/jira/browse/REEF-633)

### CODE REVIEW

**Comment by markusweimer** [comment]
Did a pass. The code looks fine, but I left some notes on the JIRA.

**Comment by afchung** [comment]
@markusweimer Updated to use a formatter instead. Please have another look. Thanks!

**Comment by markusweimer** [comment]
@afchung Thanks! This looks good, I will merge.

File: lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/DefaultRackNameFormatter.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.apache.hadoop.yarn.api.records.Container;
+import org.apache.reef.annotations.audience.DriverSide;
+import org.apache.reef.annotations.audience.Private;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Private
+@DriverSide
+public class DefaultRackNameFormatter implements RackNameFormatter {
```
Comment (markusweimer) [inline lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/DefaultRackNameFormatter.java]: Turns out that this guy needs a default constructor...

Comment (afchung) [inline lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/DefaultRackNameFormatter.java]: @markusweimer Whoops...I should probably have run tests on YARN first to make sure. Sorry!

**Pull request closed without merging**